specs/DesignDiscussions/InputRevealOnHover.md

```diff
@@ -0,0 +1,154 @@
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario7.PNG)
+
+
+6. When the user clicks on the Action.Submit or Action.Execute button, latest input values will be sent to the bot from host and bot can then update the card with latest information which we will show in default state.
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario8.PNG)
+
+## New Capabilites in the card
+In order to achieve new experience as mentioned above, we will add these capabilities in the card for developers and host:
+
+## New Schema Changes in Adaptive Card
+1. Introduce an optional styling property `inputStyle` in AC input fields schema for developer to choose between existing input fields styling i.e.`default` vs new styling i.e.`revealOnHover`.
+
+* **InputStyle: Default view:**
+
+ ![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario2.PNG)
+
+* **InputStyle: RevealOnHover View:**
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario9.PNG)
+
+
+## Use Case Clarifications:
+1. If `value` is `empty` in the input field, we will show label and placeholder text in lighter color like this in the [default] state,
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario10.PNG)
+
+which user can go and update the value.
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario11.PNG)
+
+2. If `label` is `empty`, we will just show `value` field in readable format in the default state which user can go and update the value.
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario12.PNG)
+
+
+## Developer Recommendation:
+
+1. **Save Button:** Our recommendation is to have a `Save` button (Action.Submit or Action.Execute) with `disabledUnlessAssociatedInputsChange` on the card containing inputs fields with `revealOnHover` style. Save button will send the modified user input values to the bot, and the bot will respond with a card with updated input values.
+
+2. **Cancel Button:** Our recommendation is to have a `Cancel` button (Action.Submit or Action.Execute) on the card containing inputs fields with `revealOnHover` style. If user has made some changes in the input fields but want to revert back to original state, they can click on "Cancel" button. "Cancel" button will send the card with last saved input values.
+## Out of Scope:
+
+1. Inline Action in Input.text: We will not support `revealOnHover` style for input text containing inline action. It will be shown as default input style only. Reason being, such sceanrios are reply with a comment or adding a message where user collaboration is not needed.
+
+![img](./assets/InputRevealOnHover/InputRevealOnHoverScenario13.png)
+
+2. Mobile is `out of scope` for `revealOnHover` input style design since there is no focus state on mobile and alignment will be always `vertical` as of today.
```

Comment:
we do need fallback scenarios for mobile. for example, when this card is sent to mobile users, how would the experience look like? and what's the recommended changes if any?

Reply:
Right now. this feature is out of mobile scope. But should we add parsing support in mobile and no-op implementation for this? I can take this in a subsequent PR,

Reply:
it's more about documenting how this should be handled in the mobile. if the mobile experience is as good as you have designed, no changes may be needed.